File: server/php/src/Application/Domain/Payments/PaymentDetails/PersonalName.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PersonalName
{
    public ?string $firstName;
    public ?string $surname;
    public ?string $title;

    public function __construct(?string $firstName = null, ?string $surname = null, ?string $title = null)
    {
        $this->firstName = $firstName;
        $this->surname = $surname;
        $this->title = $title;
    }

    public function toArray(): array
    {
        return [
            'firstName' => $this->firstName,
            'surname' => $this->surname,
            'title' => $this->title,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/personal_name.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PersonalName
          attr_accessor :first_name, :surname, :title

          def initialize(first_name: nil, surname: nil, title: nil)
            @first_name = first_name
            @surname = surname
            @title = title
          end

          def to_h
            {
              'firstName' => @first_name,
              'surname' => @surname,
              'title' => @title
            }
          end
        end
      end
    end
  end
end